perceval/backend.py

```diff
@@ -106,11 +106,71 @@ def archive(self, obj):
 
         self._archive = obj
 
-    def fetch(self, from_date=DEFAULT_DATETIME):
+    def fetch_items(self, **kwargs):
         raise NotImplementedError
 
-    def fetch_from_cache(self):
-        raise NotImplementedError
+    def fetch(self, category, **kwargs):
+        """Fetch items from the repository.
+
+        The method retrieves items from a repository.
+
+        :param category: the category of the items fetched
+        :param kwargs: a list of other parameters (e.g., from_date, offset, etc.
+        specific for each backend)
+
+        :returns: a generator of items
+        """
+        if self.archive:
+            self.archive.init_metadata(self.origin, Backend.__name__, self.version, category,
```

Comment:
Don't use `Backend` class. Use `self` instead. Otherwise, subclasses won't use their name. `Backend` will always be stored.